src/SourceBrowser.Site/Scripts/upload.js

```diff
@@ -0,0 +1,29 @@
+﻿$(document).ready(function () {
+    resetUI();
+    registerButtonClickEvent();
+});
+
+function registerButtonClickEvent() {
+    getButtonFromDom().click(prepareForChange);
```

Comment:
I think this prevents the default action of the button, which is to submit the form: I've set a breakpoint in `public ActionResult Submit(string githubUrl)` in `SourceBrowser.Site.Controllers.UploadController` and it never hit.

Perhaps it would be better to work with the [submit](https://api.jquery.com/submit/) event handler of the form. 
From reading the API documentation, it looks like as long as you don't set `event.preventDefault();`, the form will actually submit.